### PR TITLE
Updating some integration test components to pave the road to ORM 5.2

### PR DESCRIPTION
**Pull request by Sanne**
- https://hibernate.atlassian.net/browse/HSEARCH-2346
 - https://hibernate.atlassian.net/browse/HSEARCH-2347

These changes are extracted from my ORM 5.2 branch but can be applied on master already.

### CODE REVIEW

**Comment by yrodiere** [comment]
@Sanne So I reckon it's ready for review? Just asking because the label is missing, so I'm note absolutely sure.

**Comment by Sanne** [comment]
@yrodiere yes, please have a look and integrate it. I have more coming but this needs to go first ;-)

File: integrationtest/spring/src/test/resources/persistence.xml [hunk 1]
```diff
@@ -17,7 +17,7 @@
     </persistence-unit>
 
     <persistence-unit name="jbossjta" transaction-type="JTA">
-        <provider>org.hibernate.ejb.HibernatePersistence</provider>
+        <provider>org.hibernate.jpa.HibernatePersistenceProvider</provider>
```
Comment (yrodiere) [inline integrationtest/spring/src/test/resources/persistence.xml]: Seems like we could use the new class in `JBossTSIT.java`, too (see [here](https://github.com/hibernate/hibernate-search/blob/master/integrationtest/narayana/src/test/java/org/hibernate/search/test/integration/jbossjta/JBossTSIT.java#L88)).
Or was it on purpose that you skipped it?

**Comment by yrodiere** [comment]
@Sanne Seems good to me, apart from the missing replacement mentioned above (maybe). Feed free to merge, or tell me if you want me to take care of that replacement.

File: integrationtest/spring/src/test/resources/persistence.xml (hunk 1 above)
Comment (Sanne) [inline integrationtest/spring/src/test/resources/persistence.xml]: thanks! I had actually changed that but then sent the wrong branch .. good catch! Merging the right branch..

**Pull request closed without merging**